remote-api/src/main/java/org/alfresco/rest/api/impl/CategoriesImpl.java

```diff
@@ -114,6 +115,23 @@ public CollectionWithPagingInfo<Category> getCategoryChildren(String parentCateg
         return ListPage.of(categories, params.getPaging());
     }
 
+    @Override
+    public void deleteCategoryById(String id, Parameters params)
+    {
+        if (!authorityService.hasAdminAuthority())
+        {
+            throw new PermissionDeniedException(NO_PERMISSION_TO_DELETE_A_CATEGORY);
+        }
+
+        final NodeRef nodeRef = nodes.validateNode(id);
```

Comment:
I think we should ahave a common approach here, e.g.:
```suggestion
final NodeRef nodeRef = getCategoryNodeRef(id);
        if (isRootCategory(nodeRef))
        {
            throw new InvalidArgumentException(NOT_A_VALID_CATEGORY, new String[]{id});
        }
```